app.ts

```diff
@@ -0,0 +1,252 @@
+import { ajax, css } from "jquery";
+
+class RecordManager {
+	firstNumber: number;
+	lastNumber: number;
+	recordCount: number;
+
+	constructor() {
+		this.firstNumber = 0;
+		this.lastNumber = 0;
+		this.recordCount = 0;
+		this.initialize();
+	}
+
+	initialize() {
+		this.createTableHeader();
+		dataManager.fetchRecordCount()
+			.then(count => {
+				this.recordCount = count - 1;
+				this.updateAndDisplayRecords();
+				this.recordEventHandlers();
+				this.handleResize();
+			})
+			.catch(err => {
+				alert('Error fetching and displaying the table records, reload the page');
+				throw err;
+			});
+	}
+
+	/** Initializes the table head */
+	createTableHeader() {
+		return dataManager.fetchColumns()
+			.then(columns => {
+				for (const col of columns) {
+					$(".head").append(`<th>${col}</th>`);
+				}
+			})
+			.catch(err => {
+				alert('Error creating table heading');
+				throw err;
+			});
+	}
+
+	/** calculates the number of rows that can fit the screen */
+	calculatingRows(): number {
+		const screenHeight = window.innerHeight;
+		const availableHeight = screenHeight - 105;
+		const rowHeight = 35;
+		if (availableHeight <= 0) {
+			return 0;
+		} else {
+			let maxRows = Math.floor(availableHeight / rowHeight);
+			return maxRows;
+		}
+	}
+
+	/** fetching records that fit the screen */
+	updateAndDisplayRecords(): Promise<void> {
+		const inputValue = <string>($('#searchInput').val());
+		let errorMessage = '';
+		if (inputValue !== '') {
+		  errorMessage = 'Failed to search, reload the page and search again';
+		  $('#loader').show();
+		  this.searchRecordsAndResize();
```

Comment:
Why not call this outside of `updateAndDisplayRecords`.
Call both `this.searchRecordsAndResize` and then `this.updateAndDisplayRecords` in your search input event?